### PR TITLE
Update Test Connection Definition FQN

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

- Before: FQN was the same as name, e.g., `Mysql` or `Tableau.
- Now: FQN is generated as `Mysql.testConnectionDefinition`

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v012__create_db_connection_info.sql [hunk 1]
```diff
@@ -12,3 +12,8 @@ SET json = JSON_INSERT(
     )
 WHERE serviceType = 'Looker'
   AND JSON_EXTRACT(json, '$.connection.config.githubCredentials') IS NOT NULL;
+
+-- use FQN instead of name for Test Connection Definition
+ALTER TABLE test_connection_definition
+ADD fullyQualifiedName VARCHAR(256) GENERATED ALWAYS AS (json ->> '$.fullyQualifiedName') NOT NULL,
```
Comment (pmbrull) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v012__create_db_connection_info.sql:18]: using the name in the DAO to search for entities

File: ingestion/src/metadata/ingestion/connections/test_connections.py [hunk 2]
```diff
@@ -236,15 +236,17 @@ def test_connection_steps(
     :return: None or raise an exception if we cannot connect
     """
 
+    test_connection_def_fqn = service_type + ".testConnectionDefinition"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/connections/test_connections.py:239]: GET in the ingestion by the new FQN

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestConnectionDefinitionRepository.java [hunk 3]
```diff
@@ -31,6 +31,15 @@ public TestConnectionDefinitionRepository(CollectionDAO dao) {
         null);
   }
 
+  /**
+   * TestConnectionDefinitions are created from JSON data. The FQN will be generated out of the informed name and
+   * `.testConnectionDefinition`
+   */
+  @Override
+  public void setFullyQualifiedName(TestConnectionDefinition entity) {
+    entity.setFullyQualifiedName(entity.getName() + ".testConnectionDefinition");
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestConnectionDefinitionRepository.java:40]: Setting new FQN

File: openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx [hunk 4]
```diff
@@ -132,7 +132,10 @@ const TestConnection: FC<TestConnectionProps> = ({
 
   const fetchConnectionDefinition = async () => {
     try {
-      const response = await getTestConnectionDefinitionByName(connectionType);
+      // Test Connection FQN is built as <connectionType>.testConnectionDefinition. E.g., Mysql.testConnectionDefinition
+      const response = await getTestConnectionDefinitionByName(
+        connectionType.concat('.testConnectionDefinition')
```
Comment (pmbrull) [inline openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx:137]: updating UI GET

**Review by ulixius9 — DISMISSED** [review]

**Review by OnkarVO7 — APPROVED** [review]

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**